x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/OperatorTestCase.java

```diff
@@ -94,10 +96,18 @@ public final void testSimpleCircuitBreaking() {
         List<Page> input = CannedSourceOperator.collectPages(simpleInput(inputFactoryContext.blockFactory(), between(1_000, 10_000)));
         CircuitBreaker breaker = bigArrays.breakerService().getBreaker(CircuitBreaker.REQUEST);
         BlockFactory blockFactory = BlockFactory.getInstance(breaker, bigArrays);
-        Exception e = expectThrows(
-            CircuitBreakingException.class,
-            () -> drive(simple(bigArrays).get(new DriverContext(bigArrays, blockFactory)), input.iterator())
-        );
+        DriverContext driverContext = new DriverContext(bigArrays, blockFactory); // TODO: can replace with driverContext
+        Exception e = expectThrows(CircuitBreakingException.class, () -> {
+            Operator operator;
+            try {
+                operator = simple(bigArrays).get(driverContext);
+            } catch (CircuitBreakingException cbe) {
+                // if we failed to even create the operator, then release the input pages
+                releasePageBlocksWhileHandlingException(input);
```

Comment:
Are we safe to just close the pages without handling the exceptions?

Reply:
I reworked this so that we only close when needed, and also without suppressing exceptions.